### PR TITLE
clib: Improve Session.get_default docstring to clarify that GMT configuration parameters are supported

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Just docstring changes to clarify that `Session.get_default` can also get the **current** value of GMT configuration settings. One test is also added.

Need to note that the API name `get_default` may make people think that it returns the **default** value, but that's not true. The **current** value of a GMT configuration parameter is returned! I don't add an extra test to verify it, because the `pygmt.config` method already rely on the return value:
https://github.com/GenericMappingTools/pygmt/blob/a4d2b8e12cf1d10e31645768db0aae4253a058a3/pygmt/src/config.py#L193-L199

### CODE REVIEW

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -493,15 +495,11 @@ def get_default(self, name):
         )
 
         # Make a string buffer to get a return value
-        value = ctp.create_string_buffer(10000)
-
+        value = ctp.create_string_buffer(4096)
```
Comment (seisman) [inline pygmt/clib/session.py:498]: The `GMT_Get_Default` API function only says that `value` must have enough space to store the return value but doesn't say the exact length, but function [`gmtlib_getparameter`](https://github.com/GenericMappingTools/gmt/blob/9d48c37ef02882fd8f97673f4da35250f8c6c05f/src/gmt_init.c#L12631) (called by `GMT_Get_Default`) defines a `GMT_BUFSIZ`-long (4096) char. 

So, we can safely change 10000 to 4096 in this function.

File: pygmt/clib/session.py [hunk 2]
```diff
@@ -453,11 +453,12 @@ def destroy(self):
...
         self.session_pointer = None
 
-    def get_default(self, name):
+    def get_default(self, name: str) -> str:
         """
-        Get the value of a GMT default parameter (library version, paths, etc).
+        Get the value of a GMT configuration parameter or a GMT API parameter.
 
-        Possible default parameter names include:
+        In addtional to the long list of GMT configuration parameters, following API
```
Comment (michaelgrund) [inline pygmt/clib/session.py:460]:
```suggestion
        In addition to the long list of GMT configuration parameters, following API
```

Comment (michaelgrund) [inline pygmt/clib/session.py:460]:
```suggestion
        In addtional to the long list of GMT configuration parameters, the following API
```

**Review by michaelgrund — APPROVED** [review]

File: pygmt/clib/session.py [hunk 3]
```diff
@@ -453,11 +453,12 @@ def destroy(self):
...
         self.session_pointer = None
 
-    def get_default(self, name):
+    def get_default(self, name: str) -> str:
         """
-        Get the value of a GMT default parameter (library version, paths, etc).
+        Get the value of a GMT configuration parameter or a GMT API parameter.
 
-        Possible default parameter names include:
+        In addition to the long list of GMT configuration parameters, following API
```
Comment (seisman) [inline pygmt/clib/session.py:460]:
```suggestion
        In addition to the long list of GMT configuration parameters, the following API
```

**Pull request merged**